packages/@aws-cdk/aws-events-targets/lib/sqs.ts

```diff
@@ -52,6 +59,12 @@ export class SqsQueue implements events.IRuleTarget {
    * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/resource-based-policies-eventbridge.html#sqs-permissions
    */
   public bind(rule: events.IRule, _id?: string): events.RuleTargetConfig {
+    const role = this.props.role;
+
+    if (role !== undefined) {
+      role.addToPrincipalPolicy(this.sendMessageStatement());
```

Comment:
Why not use `this.queue.grantSendMessages(role)` ?

Reply:
The `this.queue.grantSendMessages()` call a couple lines down should probably be repurposed for this, to look like:

```ts
this.queue.grantSendMessages(this.props.role ?? new iam.ServicePrincipal('events.amazonaws.com', principalOpts));
```